### PR TITLE
Create Walkthrough One Services

### PR DESCRIPTION
**Pull request by aidenkeating**

### CODE REVIEW

File: src/common/openshiftHelpers.js [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+  });
+
+const namespaceRequestDef = {
+  name: 'projectrequests',
+  version: 'v1',
+  group: 'project.openshift.io'
+};
+const namespaceRequestResource = name => ({
+  kind: 'ProjectRequest',
+  metadata: {
```
Comment (witmicko) [inline src/common/openshiftHelpers.js]: do we need projectRequests? we can use kind project to create new namespace - https://docs.openshift.com/container-platform/3.10/rest_api/apis-project.openshift.io/v1.Project.html#Post-apis-project.openshift.io-v1-projects
check #56 openshiftKinds.js,  middlewareServices.js walkthroughServices.js for examples

File: src/services/walkthroughProvisionServices.js [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+
+/**
+ * Provisions the services required for Walkthrough 1 into a username prefixed namespace.
+ * @param {Object} amqCredentials The credentials from the AMQ instance, including URL.
+ */
+const provisionWalkthroughOne = amqCredentials =>
+  currentUser().then(user => {
+    const namespace = buildValidProjectNamespaceName(user.username, 'walkthrough-one');
+    const siDef = serviceInstanceDef(namespace);
+    return findOpenshiftResource(namespaceDef, namespaceResource(namespace))
```
Comment (JameelB) [inline src/services/walkthroughProvisionServices.js]: Can you use `findOrCreateOpenshiftResource` here?

File: src/common/openshiftHelpers.js (hunk 1 above)
Comment (aidenkeating) [reply src/common/openshiftHelpers.js]: @witmicko Have you tried running as a non-admin user? From what I remember non-admin users use ProjectRequests when creating a namespace

File: src/services/walkthroughProvisionServices.js [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+        const crudAppSiInfo = {
+          namespace,
+          name: DEFAULT_SERVICES.CRUD_APP
+        };
+        const crudSiResource = buildServiceInstanceResourceObj(crudAppSiInfo);
+        return findOrCreateOpenshiftResource(siDef, crudSiResource, buildServiceInstanceCompareFn(crudSiResource));
+      });
+  });
+
+const buildServiceInstanceCompareFn = si => res =>
```
Comment (JameelB) [inline src/services/walkthroughProvisionServices.js]: Can we move this to the service instance helper?

File: src/common/openshiftHelpers.js (hunk 1 above)
Comment (witmicko) [reply src/common/openshiftHelpers.js]: I ll that that now @aidenkeating

File: src/services/walkthroughProvisionServices.js (hunk 2 above)
Comment (aidenkeating) [reply src/services/walkthroughProvisionServices.js]: Sadly not, it's finding a `Project`, but creating a `ProjectRequest`

**Comment by aidenkeating** [comment]
@dlabaj Would you mind taking a look at this when you get the chance, tests and linters are passing.

File: src/common/serviceInstanceHelpers.js [hunk 4]
```diff
@@ -31,6 +31,18 @@ class AMQServiceInstanceTransform {
   }
 }
 
+class CRUDAppInstanceTransform {
```
Comment (dlabaj) [inline src/common/serviceInstanceHelpers.js]: Can we pull these classes out into there own files and keep them in sub directory (e.g. src/common/services/)?

Comment (dlabaj) [reply src/common/serviceInstanceHelpers.js]: Will do this in future PR.

**Pull request merged**